### PR TITLE
Speed up calling `elm-format` by not using `npx`

### PR DESCRIPTION
**Pull request by lydell**
`npx elm-format` is used to call elm-format. `npx` takes about 200 ms to boot.

This PR removes the use of `npx` for elm-format, speeding things up. This is an implementation of https://github.com/jfmengels/node-elm-review/pull/46#issuecomment-849923731 (only the elm-format bits), but done in a **backwards compatible** way. As a bonus, this gets rid of the command injection issue reported in #46 (but not the one in `elm-review new-package`).

I’ve tested manually on macOS and Windows, and it seems to work just like intended. Jeroen has tested on Linux.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -79,6 +79,7 @@
     "got": "^10.7.0",
     "minimist": "^1.2.0",
     "ora": "^5.4.0",
+    "path-key": "^3.1.1",
```
Comment (lydell) [inline package.json:82]: `cross-spawn` already depends on this module at the same version, so this is only making a sub-dependency a “real” dependency.

**Review by jfmengels — APPROVED** [review]
This looks very good! I tried it on Ubuntu and it's working very well :+1: 
In my testing, the average time went from ~1550ms to ~1250ms (sometimes even less) for a single spawn of `elm-format`, so I'm thinking your underplayed the 200ms delay :sweat_smile:

File: lib/autofix.js [hunk 2]
```diff
@@ -103,47 +104,33 @@ function checkIfAFixConfirmationIsStillExpected(app) {
...
+  if (result.error) {
+    if (result.error.code === 'ENOENT') {
+      if (hasElmFormatPathFlag) {
         throw new ErrorMessage.CustomError(
           /* eslint-disable prettier/prettier */
           'ELM-FORMAT NOT FOUND',
           `I could not find the executable for ${chalk.magentaBright('elm-format')} at the location you specified:
-  ${options.elmFormatPath}`,
+
+${options.elmFormatPath}`,
```
[the remark this answers is not in the document]
Comment (lydell) [reply lib/autofix.js:129]: It was _two_ spaces before, I thought it was a mistake 😄 

Feel free commit this change when you’ve decided on the number of spaces 👍

**Comment by lydell** [comment]
Yes, I marked as draft since I wanted to test more first. I have Windows available, but I’m too tired to test tonight.

**Comment by lydell** [comment]
I’ve tested on Windows now. Seems to work perfectly!

**Comment by jfmengels** [comment]
Awesome, thanks a lot for this!

**Pull request merged**